addons/mrp_account/models/product.py

```diff
@@ -50,7 +50,7 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves):
         dummy, bom_lines = bom.explode(self, 1)
         bom_lines = {line: data for line, data in bom_lines}
         value = 0
-        for move in stock_moves:
+        for move in stock_moves.filtered(lambda m: not m.origin_returned_move_id):
```

Comment:
in `_compute_average_price` of stock_account. the returned move must be subtract in case it is validated or not (ie there is a valuation layer linked.) 
I think we should make this distinction here too

Reply:
https://github.com/odoo/odoo/blob/0aa9925caf7a2386f00820bcc0563a22ec4ccdec/addons/stock_account/models/product.py#L665-L668